vendor/github.com/cloudevents/sdk-go/v2/event/event_reader.go

```diff
@@ -27,6 +28,26 @@ func (e Event) Type() string {
 	return ""
 }
 
+func extractScheme(sourceURL string) (string, error) {
+	parsedURL, err := url.Parse(sourceURL)
+	if err != nil {
+		return "", err
+	}
+	return parsedURL.Scheme, nil
+}
+
+// Scheme implements EventReader.Scheme
+func (e Event) Scheme() string {
+	if e.Context != nil {
+		scheme , err := extractScheme(e.Context.GetSource())
+		if err != nil {
+			return ""
+		}
+		return scheme
+	}
+	return ""
+}
+
```

Comment:
@sadath-12 these are vendored dependencies, so any changes you make will just be overriden the next time the dependencies are updated. In general, you should not make changes to the `vendor` directory. Would you be able to refactor your code to not modify these files?

Reply:
sure thanks for letting me know